keps/sig-multicluster/4322-cluster-inventory/README.md

```diff
@@ -0,0 +1,1021 @@
+  manner.
+
+### Terminology
+
+- **Cluster Manager**: An entity that creates the cluster inventory API 
+  resources per member cluster, and keeps their status up-to-date. Each
+  cluster manager SHOULD be identified with a unique name. Each cluster
+  inventory resource SHOULD be managed by only one cluster manager. Examples
+  of cluster manager are projects like OCM, Karmada, Clusternet or Azure
+  fleet manager.
+
+- **Cluster Inventory Consumer**: the person running the cluster managers
+  or the person developing extensions for cluster managers for the purpose of
+  workload distribution, operation management etc.
+
+- **Member Cluster**: A kubernetes cluster that is managed by the cluster
+  manager. A cluster manager SHOULD have sufficient permission to access
+  the member cluster to fetch the information so it can update the status
+  of the cluster inventory API resource.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1: Multicluster Workload Distribution
+
+In this scenario, a standard API acts as a foundation for multicluster
+scheduling decisions across various clusters. This means that the
+API will provide a single point of contact to retrieve information that can be
+used to make informed scheduling decisions for workloads across multiple
+clusters. For instance, workload distribution tools like GitOps or
+[Work API](https://github.com/kubernetes-sigs/work-api) can leverage this
+API to make informed decisions about which cluster is best suited to handle
+a particular workload.
+
+Examples of the multicluster scheduling includes:
+
+- As a user I want to run a workload on an EKS cluster that resides in us-east-1.
+  I want to submit my workload if and only if a cluster satisfies that constraint.
+- As a user I want to run a workload on a cluster that has Kueue/Yunikorn enabled.
+  The cluster level scheduling wants to submit a workload on a cluster that has the
+  shortest time to being picked up.
+- As a user I want to run a workload on a cluster that has certain CRDs installed.
+- As a user I want to deploy a workload to my on-prem cluster group.
+- As a user I want to run a workload close to the data source or end-users.
+- As a user I want to run a workload to the less busy cluster.
+
+#### Story 2: Operations and Management
+
+For multi-cluster administrators, the unified API provides a
+comprehensive view of all the clusters under their management. This
+includes verifying their memberships, understanding their status,
+capacity, and healthiness.
+
+For instance, the API could provide real-time data about the CPU usage,
+memory consumption, network latency, and other key performance
+indicators of each cluster. If a cluster is nearing its capacity or
```

Comment:
Same concerns here - this kind of data can be very high traffic, what are the implications to scalability?

Is this API the right place for this sort of data? There are monitoring solutions that are built specifically to handle metrics like these, that really should accompany any production deployment already and support the sort of alerting we'd want here. Can we hand off to them?

Reply:
yes, removed this part and add resource based scheduling as beta plan.